tests/unittests/cmd/test_main.py

```diff
@@ -129,58 +120,15 @@ def test_main_init_run_net_calls_set_hostname_when_metadata_present(self):
         )
 
         def set_hostname(name, cfg, cloud, args):
-            self.assertEqual("set_hostname", name)
-            updated_cfg = copy.deepcopy(self.cfg)
-            updated_cfg.update(
-                {
-                    "def_log_file": "/var/log/cloud-init.log",
-                    "log_cfgs": [],
-                    "syslog_fix_perms": [
-                        "syslog:adm",
-                        "root:adm",
-                        "root:wheel",
-                        "root:root",
-                    ],
-                    "vendor_data": {"enabled": True, "prefix": []},
-                    "vendor_data2": {"enabled": True, "prefix": []},
-                }
-            )
-            updated_cfg.pop("system_info")
-
-            self.assertEqual(updated_cfg, cfg)
-            self.assertIsNone(args)
-
-        (_item1, item2) = wrap_and_call(
-            "cloudinit.cmd.main",
-            {
-                "close_stdin": True,
-                "netinfo.debug_info": "my net debug info",
-                "cc_set_hostname.handle": {"side_effect": set_hostname},
-                "util.fixup_output": ("outfmt", "errfmt"),
-            },
-            main.main_init,
-            "init",
-            cmdargs,
-        )
-        self.assertEqual([], item2)
-        # Instancify is called
-        instance_id_path = "var/lib/cloud/data/instance-id"
-        self.assertEqual(
-            "iid-datasource-none\n",
-            os.path.join(
-                load_text_file(os.path.join(self.new_root, instance_id_path))
-            ),
-        )
-        # modules are run (including write_files)
-        self.assertEqual(
-            "blah", load_text_file(os.path.join(self.new_root, "etc/blah.ini"))
+            assert "set_hostname" == name
```

Comment:
I did simplify the logic here and remove a lot of extraneous checks in the rest of the test. The goal of this test was to ensure that `set_hostname` gets called from main.py, and the rest of the checks that happened here already exist in the previous test.